compiler/optimizations/gpuTransforms.cpp

```diff
@@ -1945,26 +2068,6 @@ static VarSymbol* generateNumThreads(BlockStmt* gpuLaunchBlock,
   return numThreads;
 }
 
-void GpuKernel::buildStubKernelLaunchBlock() {
-  launchBlock_ = new BlockStmt();
-
-  std::vector<BlockStmt*> blocksInBody;
-  collectBlockStmts(gpuLoop.loop(), blocksInBody);
-
-  for (auto block : blocksInBody) {
-    if (block->isGpuPrimitivesBlock()) {
-      processGpuPrimitivesBlock(block);
-    }
-  }
-
-  // we are done with the call, now remove it.
-  if (auto bsc = blockSizeCall()) {
-    if (bsc->inTree()) {
-      bsc->remove();
-    }
-  }
```

Comment:
The removal is now at the end of `processGpuPrimitivesBlock()`. It was actually there before my PR, except it used curlies and `toCallExpr(expr)` instead of `call`:
```c++
      if (isCallToPrimitiveWeShouldNotCopyIntoKernel(call))
        call->remove();
```